packages/babel-types/test/converters.js

```diff
@@ -320,7 +320,10 @@ describe("converters", function () {
     it("gathers empty statements", function () {
       const node = parseCode(";");
       const sequence = t.toSequenceExpression([undefinedNode, node], scope);
-      expect(generateCode(sequence.expressions[1])).toBe("undefined");
+      expect(() => {
+        generateCode(sequence.expressions[1]);
+      }).toThrow("Cannot read property '1' of undefined");
+      // expect(generateCode(sequence.expressions[1])).toBe("undefined");
```

Comment:
The test is failing because the test assumption does not hold if the empty statement is following a value producing statement, in this test case, `undefined ;` will _not_ produce a sequence expression because the empty statement is not the leading statement. Therefore `sequence.expression` is `undefined`.

However, asserting an error thrown because it is in fact _not_ an sequence expression can be confusing to other contributors. Moreover, we know that it still _gathers_ empty statement only if the `EmptyStatement` is the first element:

So we need to revise the test and clearly state its new behaviour

```js
it("gathers leading empty statement", function() {
  const node = parseCode(";");
  const sequence = t.toSequenceExpression([node, t.cloneNode(node)], scope);
  expect(generateCode(sequence)).toBe("undefined");
});
```

Reply:
Oh I see, also is the difference between `toBeUndefined()` vs. `toBe("undefined")`  just the type it returns?